annnufan/assignment1/solution_2.cpp

```diff
@@ -0,0 +1,65 @@
+#include <string>
+#include <iostream>
+
+template <typename T>
+class node;

```

Comment:
I don't think there is a need to declare `node` here; a class can have a reference to itself without pre-declaring it.